dts/msm8916/msm8916-512mb-mtp.dts

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton.dtsi>
+#include <lk2nd.h>
+
+/ {
+	compatible = "qcom,msm8916-512mb-mtp", "qcom,msm8916";
+	qcom,board-id = <8 0x100>;
+	
+	zhihe-ufi-001c {
+		model = "ufi-001b/ufi-001c 4G Modem Stick";
+		compatible = "zhihe,ufi-001c", "qcom,msm8916", "lk2nd,device";
+		qcom,msm-id = <206 0>;
+
+		/* Register EDL button as HOME button */
+		/* This is the only button on this device */
+		lk2nd,keys = <KEY_HOME 37 (GPIO_PULL_DOWN | GPIO_ACTIVE_HIGH)>;
+	};
+	
+	zhihe-sp970 {
+		model = "sp970 4G Modem Stick";
+		compatible = "zhihe,sp970", "qcom,msm8916", "lk2nd,device";
+		qcom,msm-id = <248 0>;
```

Comment:
`206` is fine. Besides `248` stands for MSM8**2**16, which is 3G only, but this is a 4G modem.

https://github.com/msm8916-mainline/lk2nd/blob/0.14.0/dts/msm8916/msm8216-samsung-r05.dts